pandas/tests/frame/methods/test_replace.py

```diff
@@ -1632,3 +1632,37 @@ def test_replace_unicode(self):
         result = df1.replace(columns_values_map)
         expected = DataFrame({"positive": np.ones(3)})
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "to_replace,value,input_data,expected_data,inplace",
+        [
+            (r"^\s*$", pd.NA, ["d", "ee", "f", ""], ["d", "ee", "f", pd.NA], False),
+            (r"e{2}", "rep", ["d", "ee", "f", ""], ["d", "rep", "f", ""], False),
+            (r"f", "replace", ["d", "ee", "f", ""], ["d", "ee", "replace", ""], False),
+            (r"^\s*$", pd.NA, ["d", "ee", "f", ""], ["d", "ee", "f", pd.NA], True),
+            (r"e{2}", "replace", ["d", "ee", "f", ""], ["d", "replace", "f", ""], True),
+            (r"f", "replace", ["d", "ee", "f", ""], ["d", "ee", "replace", ""], True),
+        ],
+    )
+    def test_replace_regex(self, to_replace, value, input_data, expected_data, inplace):
+        # GH35977
+        df = pd.DataFrame({"col1": input_data}, dtype="string")
+        expected = pd.DataFrame({"col1": expected_data}, dtype="string")
+        df_replaced = df.replace(to_replace, value, inplace=inplace, regex=True)
+        result = df if inplace else df_replaced
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "to_replace,value,input_data,expected_data",
+        [
+            ("", pd.NA, ["d", "ee", "f", ""], ["d", "ee", "f", pd.NA]),
+            ("ee", "replace", ["d", "ee", "f", ""], ["d", "replace", "f", ""]),
+            ("f", "replace", ["d", "ee", "f", ""], ["d", "ee", "replace", ""]),
+        ],
+    )
+    def test_replace_string(self, to_replace, value, input_data, expected_data):
```

Comment:
can you parametrize these using frame_or_series

Reply:
was this addressed?